frontend/src/views/WheresMyGeneV2/components/InfoPanel/components/Legend/connect.ts

```diff
@@ -0,0 +1,29 @@
+import { useCallback, useState } from "react";
+import { useFilterDimensions } from "src/common/queries/wheresMyGene";
+import { EVENTS } from "src/common/analytics/events";
+import { track } from "src/common/analytics";
+import { COLOR_LEGEND } from "../RelativeGeneExpression/constants";
+
+export const useConnect = () => {
+  const [hoverStartTime, setHoverStartTime] = useState(0);
+  const useHandleHoverEnd = (event: EVENTS, payload = {}) => {
+    return useCallback(() => {
+      if (Date.now() - hoverStartTime > 2 * 1000) {
+        track(event, payload);
+      }
+    }, [event, payload]);
+  };
+
+  const handleLegendHoverEnd = useHandleHoverEnd(
+    EVENTS.WMG_LEGEND_QUESTION_BUTTON_HOVER,
+    { label: COLOR_LEGEND }
+  );
+
+  const { data: filterDimensions } = useFilterDimensions();
+
+  const { datasets = [] } = filterDimensions;
```

Comment:
let's use `EMPTY_ARRAY` to prevent creating a new `[]` instance every time!

Reply:
Not sure why this happened again. But this comment hasn't been addressed but somehow it's resolved. Reopening!